core/main/src/firebolt/handlers/hdmi_rpc.rs

```diff
@@ -0,0 +1,198 @@
+    },
+    firebolt::{
+        fb_general::{ListenRequest, ListenerResponse},
+        panel::fb_hdmi::{
+            GetAvailableInputsResponse, HdmiOperation, HdmiSelectOperationRequest,
+            HdmiSelectOperationResponse,
+        },
+    },
+    gateway::rpc_gateway_api::CallContext,
+};
+use ripple_sdk::{
+    api::device::device_events::AUTO_LOW_LATENCY_MODE_SIGNAL_CHANGED, log::error, serde_json,
+};
+use ripple_sdk::{
+    api::device::panel::device_hdmi::HdmiRequest,
+    extn::extn_client_message::{ExtnPayload, ExtnResponse},
+};
+
+#[rpc(server)]
+pub trait Hdmi {
+    #[method(name = "HDMIInput.select")]
+    async fn select_hdmi_operation(
+        &self,
+        ctx: CallContext,
+        request: HdmiSelectOperationRequest,
+    ) -> RpcResult<HdmiSelectOperationResponse>;
+
+    #[method(name = "HDMIInput.ports")]
+    async fn get_available_inputs(&self, ctx: CallContext)
+        -> RpcResult<GetAvailableInputsResponse>;
+
+    #[method(name = "HDMIInput.onConnectionChanged")]
+    async fn on_hdmi_connection_changed(
+        &self,
+        ctx: CallContext,
+        request: ListenRequest,
+    ) -> RpcResult<ListenerResponse>;
+
+    #[method(name = "HDMIInput.onAutoLowLatencyModeSignalChanged")]
+    async fn on_auto_low_latency_mode_signal_changed(
+        &self,
+        ctx: CallContext,
+        request: ListenRequest,
+    ) -> RpcResult<ListenerResponse>;
+}
+
+#[derive(Debug)]
+pub struct HdmiImpl {
+    pub state: PlatformState,
+}
+
+#[async_trait]
+impl HdmiServer for HdmiImpl {
+    async fn select_hdmi_operation(
+        &self,
+        _ctx: CallContext,
+        request: HdmiSelectOperationRequest,
+    ) -> RpcResult<HdmiSelectOperationResponse> {
+        if let HdmiOperation::Start = request.operation {
+            if let Ok(response) = self
+                .state
+                .get_client()
+                .send_extn_request(HdmiRequest::StartHdmiInput(request))
+                .await
+            {
+                if let ExtnPayload::Response(ExtnResponse::Value(value)) = response.payload {
+                    if let Ok(res) = serde_json::from_value::<HdmiSelectOperationResponse>(value) {
+                        return Ok(res);
+                    }
+                }
+            }
+
+            Err(rpc_err("FB error response TBD"))
+        } else {
+            if let Ok(response) = self
+                .state
+                .get_client()
+                .send_extn_request(HdmiRequest::StopHdmiInput(request))
+                .await
+            {
+                if let ExtnPayload::Response(ExtnResponse::Value(value)) = response.payload {
+                    if let Ok(res) = serde_json::from_value::<HdmiSelectOperationResponse>(value) {
+                        return Ok(res);
+                    }
+                }
+            }
+
+            Err(rpc_err("FB error response TBD"))
+        }
```

Comment:
The if/else block branches look the same to me. Is that right?

Reply:
Yes needs some cleanup by passing the operation to the Device handler

Reply:
Fixed it in latest commit https://github.com/rdkcentral/Ripple/pull/264/commits/01ea44059d1e605b352282165ba95169a72d7e52#diff-130906b50b27c33d5f205f3da7521a976f98012695e1c65d5882619ddf96973fR73